### DISCUSSION TITLE
Help with RLS

### DISCUSSION THREAD

**Discussion by pr0t0n-cmd** [supabase/supabase#22336] [Questions] [answered]
Hello, i am having trouble running this function, it gives me stack depth limit exceeded (i have tried also other queries)

  select 1
  from
    points p
    join profiles pt on p.user_id = pt.id
  where
    agency_id = auth.uid()

i have the profiles table, then i have another table called points, in this table i have associated every "point". i have a column called user_id, that contains a link to a profile ID and then i have an agency_id that also have a link with the profiles table ID,
The goal is to be able to only see pertinent profiles. when i login as agency, i want to use that auth.uid() and to be able to see every other profile that, in the points table, has the column agency_id that is the same as the auth.uid() (of the agency). the current rls on the profiles table is a id = auth_id() for ALL and i have a policy on the points table that has the same rule but the agency_id = auth.uid().

I am clearly missing something, any help appreciated thanks

**Comment by GaryAustin1** [comment, marked as answer]
The error almost always means you have recursion going on because you are setting a select policy on a table, but need to select the table in the policy.  To solve this you need to move the select code in the policy to a 'security definer' type function to do the select so it bypass RLS.

https://supabase.com/docs/guides/auth/row-level-security#using-security-definer-functions

**Reply by pr0t0n-cmd** [reply]
Ok thanks I'll try later, can you tell me what's the difference between invoker/definer?

**Reply by GaryAustin1** [reply]
Security definer runs as the owner versus the user.  The owner is usually postgres role when you create the function in the dashboard.  Postgres role ignores RLS.
